releasenotes/notes/gomemlimit.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+- 40676
+
+releaseNotes:
+- |
+  **Added** an automatically set GOMEMLIMIT to `istiod` to reduce the risk of out-of-memory issues.
```

Comment:
IMO, GOMEMLIMIT is not really a tool to reduce the out-of-memory issues. If you're out of memory, you are less likely to OOM, but instead, you'll see performance degradation - as you get closer to the limit, GC will try harder not to breach it. So you're trading one issue for another.

GOMEMLIMIT is, however, a great CPU utilization reduction tool for services that are very far from the "physical" memory limit though - you'll run GC less often.